packages/@expo/config-plugins/src/utils/validations.ts

```diff
@@ -0,0 +1,72 @@
+// this is an approximation of the android resource name rules
+const validPattern = /^[a-z][a-z0-9_]*$/;
+
+const blockList = [
+  'abstract',
+  'continue',
+  'for',
+  'new',
+  'switch',
+  'assert',
+  'default',
+  'if',
+  'package',
+  'synchronized',
+  'boolean',
+  'do',
+  'goto',
+  'private',
+  'this',
+  'break',
+  'double',
+  'implements',
+  'protected',
+  'throw',
+  'byte',
+  'else',
+  'import',
+  'public',
+  'throws',
+  'case',
+  'enum',
+  'instanceof',
+  'return',
+  'transient',
+  'catch',
+  'extends',
+  'int',
+  'short',
+  'try',
+  'char',
+  'final',
+  'interface',
+  'static',
+  'void',
+  'class',
+  'finally',
+  'long',
+  'strictfp',
+  'volatile',
+  'const',
+  'float',
+  'native',
+  'super',
+  'while',
+  '_',
+];
+
+export function isAndroidAssetNameValid(assetNameWithoutExtension: string) {
```

Comment:
```suggestion
export function isValidAndroidAssetName(assetNameWithoutExtension: string) {
```

i know the name is coming from existing code. maybe we can rename it and align your `assertValidAndroidAssetName` naming convention